### ISSUE TITLE
Unable to delete products - "Variant is missing its product"

### ISSUE THREAD

**Issue by mattsbanner** [craftcms/commerce#1718]
**Description**
Unable to delete products through the CP. These are not the default products, so is likely a separate issue to https://github.com/craftcms/commerce/issues/358.

The CP doesn't show an error, but the post request to `/index.php?p=admincp/actions/element-indexes/perform-action&v=1599666479479` returns a 500 with "Variant is missing its product".

**To Reproduce**
How, and what happened?
1. Create a product
2. Delete the product
3. Error

**Expected Behaviour**
Product deleted.

**Additional info**
- Craft version: 3.4.30
- Commerce version: 3.1.12
- PHP version: 7.3
- Database driver & version: MySQL 5.7
- Other Plugins & versions: Stripe & PayPal for Commerce

**Additional context**
This is happening on products with and without variants.

**Comment by nfourtythree** [comment]
Hi @mattsbanner 

I have just tried this on a fresh up-to-date install and everything works as expected.

Have you tried updating Commerce and Craft to the latest versions to see if the issue persists?

Thanks.

**Comment by shornuk** [comment]
@mattsbanner  We were having the same issue. Have you manually deleted any products from the database? You'll need to make sure everything related to that product is deleted, i.e Variants, Purchasable's and elements. Once these were cleared everything deleted as normal.

**Comment by mattsbanner** [comment]
Hi @nfourtythree, @shornuk 

I have tried updating Craft and all plugins but unfortunately still received the same error.

This isn't a site we built, but one that we now maintain from a previous developer. So it's entirely possible this issue could be a result of a database that's been fiddled with.

I'm currently waiting on some further time approval from the client before I can look into this more, so happy for the issue to be closed. Sounds like some investigation within the database could probably solve the issue anyway!

Thanks.
